src/generateElementCreator.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import Relay from 'react-relay';
+
+import * as Utils from './containerUtils';
+import QueryAggregator from './QueryAggregator';
+import getRouteParams from './getRouteParams';
+import RouteGenerator from './RouteGenerator';
+import getRelayRootProps from './getRelayRootProps';
+
+export default function generateElementCreator() {
+  const queryAggregator = new QueryAggregator();
+  const routeGenerator = new RouteGenerator();
+
+  return function createElement(Component, props) {
+    const {branch, route, params, location, routeParams} = props;
+
+    // Wrap any Relay.Containers
+    let element = !Relay.isContainer(Component) ? <Component {...props} /> :
+      Utils.createContainerElement(
+        Component,
+        props,
+        getRouteParams(routeParams, location.query, route.queryParams),
+        getRelayRootProps(route),
```

Comment:
Maybe we can be more explicit here and just have e.g. a `rootContainerProps` property on the route - it's quite a lot uglier to use, but makes the API more future-proof.